Combinatorial_map/include/CGAL/Face_graph_wrapper.h

```diff
@@ -0,0 +1,944 @@
+// Copyright (c) 2019 CNRS and LIRIS' Establishments (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org); you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public License as
+// published by the Free Software Foundation; either version 3 of the License,
+// or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0+
+//
+// Author(s)     : Guillaume Damiand <guillaume.damiand@liris.cnrs.fr>
+//
+#ifndef CGAL_FACE_GRAPH_WRAPPER_H
+#define CGAL_FACE_GRAPH_WRAPPER_H 1
+
+#include <CGAL/Functors_for_face_graph_wrapper.h>
+#include <CGAL/Iterators_for_face_graph_wrapper.h>
+#include <CGAL/internal/Combinatorial_map_internal_functors.h>
+
+namespace CGAL
+{
+  // Forward declarations of all classes model of Face_graph
+  template <unsigned int d, typename Refs, typename Items, typename Alloc,
+            typename Storage>
+  class Combinatorial_map_base;
+  
+  template <unsigned int d, typename Refs, typename Items, typename Alloc,
+            typename Storage>
+  class Generalized_map_base;
+  
+  template <unsigned int d, unsigned int d2, typename Traits, typename Items,
+            typename Alloc,
+            template<unsigned int,class,class,class,class>
+            class Map, typename Refs, typename Storage>
+  class Linear_cell_complex_base;
+
+  template <unsigned int d, typename Items, typename Alloc,
+            typename Storage>
+  class Combinatorial_map;
+
+  template <unsigned int d, typename Items, typename Alloc,
+            typename Storage>
+  class Generalized_map;
+
+  template <unsigned int d, unsigned int d2, typename Traits, typename Items,
+            typename Alloc,
+            template<unsigned int,class,class,class,class>
+            class Map, typename Storage>
+  class Linear_cell_complex_for_combinatorial_map;
+  
+  template <unsigned int d, unsigned int d2, typename Traits, typename Items,
+            typename Alloc,
+            template<unsigned int,class,class,class,class>
+            class Map, typename Storage>
+  class Linear_cell_complex_for_generalized_map;
+
+  template<class P>
+  class Surface_mesh;
+
+  template<class PolyhedronTraits_3,
+           class PolyhedronItems_3,
+           template<class T, class I, class A> class T_HDS,
+           class Alloc>
+  struct Polyhedron_3;
```

Comment:
Forward-declarations are not robust to refactoring. Include `<CGAL/Polyhedron_3_fwd.h>` and `<CGAL/Surface_mesh/Surface_mesh_fwd.h>` instead. They are in `Installation/include/` to avoid any trouble with our dependencies checker.

Reply:
Ok, thanks !